proxy/src/main/java/com/velocitypowered/proxy/util/ClosestLocaleMatcher.java

```diff
@@ -54,4 +61,43 @@ public void registerKnown(final Locale locale) {
   public Locale lookupClosest(final Locale locale) {
     return closest.get(locale);
   }
+
+  public static Component translateAndParse(String key, Object... arguments) {
+    return translateAndParse(key, null, arguments);
+  }
+
+  public static Component translateAndParse(String key, Locale locale, Object... arguments) {
+    return translateAndParse(key, locale, NamedTextColor.WHITE, arguments);
+  }
+
+  /**
+   * Translate and parse with MiniMessage an translation.
+   * @param key the translation key
+   * @param locale the locale
+   * @param color the color to set if the resulting component does not have color
+   * @param arguments the arguments
+   * @return the translation parsed
+   */
+  public static Component translateAndParse(String key, Locale locale, TextColor color, Object... arguments) {
+    return MiniMessage.miniMessage()
+      .deserialize(GlobalTranslator.translator()
+        .translate(key, INSTANCE.lookupClosest(locale == null
+          ? Locale.getDefault()
+          : locale
+        ))
+        .format(format(arguments), new StringBuffer(), null).toString())
+      .colorIfAbsent(color);
+  }
+
+  private static final Object[] format(Object... arguments) {
+    if (arguments == null || arguments.length == 0) {
+      return arguments;
+    }
+    for (int i = 0; i <= 0; i++) {
```

Comment:
This does only work when you're providing exactly one argument.
Your ArrayIndexOutOfBoundException is related to that: The correct iteration for an array is `for (int i = 0; i < array.length; i++)`

Reply:
Thanks for commenting, I really didn't notice that mistake, thank you very much

Reply:
Now you'll receive every time a ArrayIndexOutOfBoundException...

Reply:
I made the change without seeing it much, thanks again